systemtests/src/test/java/com/github/streamshub/systemtests/ConsumerST.java

```diff
@@ -0,0 +1,130 @@
+
+import com.github.streamshub.systemtests.constants.Constants;
+import com.github.streamshub.systemtests.enums.ResetOffsetDateTimeType;
+import com.github.streamshub.systemtests.enums.ResetOffsetType;
+import com.github.streamshub.systemtests.locators.ConsumerGroupsPageSelectors;
+import com.github.streamshub.systemtests.logs.LogWrapper;
+import com.github.streamshub.systemtests.setup.console.ConsoleInstanceSetup;
+import com.github.streamshub.systemtests.setup.strimzi.KafkaSetup;
+import com.github.streamshub.systemtests.utils.playwright.PwPageUrls;
+import com.github.streamshub.systemtests.utils.playwright.PwUtils;
+import com.github.streamshub.systemtests.utils.resourceutils.KafkaClientsUtils;
+import com.github.streamshub.systemtests.utils.resourceutils.KafkaCmdUtils;
+import com.github.streamshub.systemtests.utils.resourceutils.KafkaNamingUtils;
+import com.github.streamshub.systemtests.utils.resourceutils.ResourceUtils;
+import com.github.streamshub.systemtests.utils.testutils.ConsumerTestUtils;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.skodjob.testframe.TestFrameConstants;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ConsumerST extends AbstractST {
+    private static final Logger LOGGER = LogWrapper.getLogger(ConsumerST.class);
+
+    /**
+     * Tests resetting Kafka consumer group offsets across multiple topics using various offset reset types via the UI.
+     *
+     * <p>The test first prepares the scenario by creating topics and producing/consuming messages to set initial offsets.</p>
+     * <p>It verifies the default consumer offsets correspond to the latest message offsets before any reset.</p>
+     * <p>The test then navigates to the Consumer Groups page and uses the UI to reset offsets for all topics using:</p>
+     * <ul>
+     *   <li>Earliest offset reset</li>
+     *   <li>Latest offset reset</li>
+     *   <li>Date/time-based earliest latest offset resets with UNIX epoch format</li>
+     *   <li>Date/time-based middle offset resets with ISO-8601 format</li>
+     * </ul>
+     * <p>For each reset type, the test verifies that the consumer group offsets are updated accordingly by querying Kafka through the broker pod.</p>
+     * <p>The test ensures that both dry-run and actual reset operations work as expected and that offsets reflect the UI commands.</p>
+     *
+     * <p>This comprehensive test validates the correctness of consumer offset reset functionality across multiple topics and offset types,
+     * ensuring synchronization between UI actions and Kafka consumer group state.</p>
+     */
+    @Test
+    void testResetConsumerOffsetAllTopicsVariousOffsets() {
```

Comment:
Any benefit to splitting this to be a `@ParameterizedTest`? I don't know how heavy the setup/teardown is for each test, but if it's low then having it be parameterized would pinpoint failures to a single offset type/value. Not required for this PR as having some tests in place is more important and they can be refined later.

Reply:
Yes, right now `@test`'s `@beforeEach` consists of:  create namespace  -> deploy kafka -> deploy console -> login -> start testing. It certainly is a concern. For test that's in this PR it deploys kafka only once. There are some cases where kafka needs to be fresh as it's faster than figuring out what went wrong in previously failed test and re-deploying/cleaning only that part of the kafka.